### PR TITLE
chore(MONGOSH-1215): Test for a token in the new format

### PR DESCRIPTION
**Pull request by lerouxb**

### CODE REVIEW

File: test/index.ts [hunk 1]
```diff
@@ -108,4 +108,26 @@ describe('decodeResumeToken', function() {
       documentKey: null
     });
   });
+
+  // See PM-1950
+  it('can decode new resume tokens', async() => {
+    const decoded = decodeResumeToken('826275077A000000012B042C0100296E5A1004BFFDB617400E486099007C0E0048B305463C6F7065726174696F6E54797065003C696E736572740046646F63756D656E744B65790046645F696400646275077A2F3159F971E405C6000004');
```
Comment (addaleax) [inline test/index.ts:114]: Does this mean that this just works? What really changed here then? :)

**Review by addaleax — APPROVED** [review]

File: test/index.ts (hunk 1 above)
Comment (lerouxb) [reply test/index.ts:114]: https://mongodb.slack.com/archives/CKABWR2CT/p1651763241585389?thread_ts=1651760108.617669&cid=CKABWR2CT

I don't know if the explicit uuid null thing needs to be tested and even how to get a token like that. looking at maybeToUUID it seems to deal with that case, though. Although maybe now better to return null than undefined?

Arguably what we call documentKey should now be named eventIdentifier? It seems to be positional and then we name it what it is.

There's definitely an argument to be made to wait for 6.1 and more specs/examples..

Comment (addaleax) [reply test/index.ts:114]: > Arguably what we call documentKey should now be named eventIdentifier?

Yes, I think so, for v2 resume tokens we should probably do that :)

> Although maybe now better to return null than undefined?

I don’t think this matters much, tbh :woman_shrugging:

Comment (lerouxb) [reply test/index.ts:114]: Done. What do you think?

**Pull request merged**